### PR TITLE
add evaluate over trace

### PR DESCRIPTION
**Pull request by ferrine**
Following ideas from https://discourse.pymc.io/t/make-predictions-with-the-particles-of-svgd/2040/5

### CODE REVIEW

File: pymc3/variational/approximations.py [hunk 1]
```diff
@@ -570,6 +570,26 @@ def __init__(self, trace=None, size=None, **kwargs):
             raise opvi.LocalGroupError('Empirical approximation does not support local variables')
         super(Empirical, self).__init__(trace=trace, size=size, **kwargs)
 
+    def evaluate_over_trace(self, node):
```
Comment (ferrine) [inline pymc3/variational/approximations.py:573]: I do not like the name very much, another candidate was `integrate` but it does not show what is actually going on

Comment (springcoil) [reply pymc3/variational/approximations.py:573]: Is it calculating expectations over the trace?

Comment (ferrine) [reply pymc3/variational/approximations.py:573]: No, I leave that for user. He may need variance or other stuff

Comment (twiecki) [reply pymc3/variational/approximations.py:573]: Variance is an expectation :).

Comment (junpenglao) [reply pymc3/variational/approximations.py:573]: How about:
```suggestion
    def monte_carlo_expectation(self, node):
```

Comment (ferrine) [reply pymc3/variational/approximations.py:573]: I do not like it. Even if variance is computed as an expectation that is not practical to manually pass the correct expression to integrate over the trace.

Comment (ferrine) [reply pymc3/variational/approximations.py:573]: currently the method just evaluates node over the trace and returns every evaluation

Comment (junpenglao) [reply pymc3/variational/approximations.py:573]: Oh ok, then `evaluate_over_trace` is good (or maybe `evaluate_over_points`?)

File: pymc3/variational/approximations.py [hunk 2]
```diff
@@ -570,6 +570,26 @@ def __init__(self, trace=None, size=None, **kwargs):
...
+        R"""
+        This allows to statically integrate over the trace any symbolic expression.
```
Comment (twiecki) [inline pymc3/variational/approximations.py]:
```suggestion
        This allows to statically evaluate any symbolic expression over the trace.
```

File: pymc3/variational/approximations.py (hunk 1 above)
Comment (ferrine) [reply pymc3/variational/approximations.py:573]: I think trace is more intuitive

**Comment by twiecki** [comment]
Thanks @ferrine!

**Pull request merged**